extractors/nwbextractors/nwbextractors.py

```diff
@@ -68,6 +69,106 @@ def get_dynamic_table_property(dynamic_table, *, row_ids=None, property_name):
     return [dynamic_table[property_name][all_row_ids.index(x)] for x in row_ids]
 
 
+class NwbImagingExtractor(ImagingExtractor):
+    '''
+        Class used to extract data from the NWB data format. Also implements a
+        static method to write any format specific object to NWB.
+        '''
+
+    extractor_name = 'NwbImaging'
+    installed = HAVE_NWB # check at class level if installed or not
+    is_writable = True
+    mode = 'file'
+    installation_mesg = "To use the Nwb Extractor run:\n\n pip install pynwb\n\n"  # error message when not installed
+
+    def __init__(self, filepath, optical_channel_name=None,
+                 imaging_plane_name=None, image_series_name=None,
+                 processing_module_name=None,
+                 neuron_roi_response_series_name=None,
+                 background_roi_response_series_name=None):
+        '''
+        Parameters
+        ----------
+        filepath: str
+            The location of the folder containing dataset.nwb file.
+        optical_channel_name: str(optional)
+            optical channel to extract data from
+        imaging_plane_name: str(optional)
+            imaging plane to extract data from
+        image_series_name: str(optional)
+            imaging series to extract data from
+        processing_module_name: str(optional)
+            processing module to extract data from
+        neuron_roi_response_series_name: str(optional)
+            name of roi response series to extract data from
+        background_roi_response_series_name: str(optional)
+            name of background roi response series to extract data from
+        '''
+        assert HAVE_NWB, self.installation_mesg
+        ImagingExtractor.__init__(self)
+
+    #TODO placeholders
+    def get_frame(self, frame_idx):
+        assert frame_idx < self.get_num_frames()
+        return self._video[frame_idx]
+
+    def get_frames(self, frame_idxs):
+        assert np.all(frame_idxs < self.get_num_frames())
+        planes = np.zeros((len(frame_idxs), self._size_x, self._size_y))
+        for i, frame_idx in enumerate(frame_idxs):
+            plane = self._video[frame_idx]
+            planes[i] = plane
+        return planes
```

Comment:
why not just 

```python
return self._video[frame_idxs]
```

if you are worried about non-ascending indices (a constraint of h5py), you could sort them first and then unsort them

```python
sorted_frame_idxs, sorting_inverse = np.sort(frame_idxs, return_inverse=True)
return self._video[sorted_frame_idxs][sorting_inverse]
```

Reply:
Yep that should work in most cases